---
description: 'Дополнительные сведения о: C26110'
title: C26110
ms.date: 10/01/2019
ms.topic: reference
f1_keywords:
- C26110
helpviewer_keywords:
- C26110
ms.assetid: d82b79ec-6d7f-438b-bd6a-da874a3e08e5
ms.openlocfilehash: d1532e2b61e80da98eedc35f09d2483d723fcb86
ms.sourcegitcommit: d6af41e42699628c3e2e6063ec7b03931a49a098
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 12/11/2020
ms.locfileid: "97279277"
---
# <a name="c26110"></a>C26110

> предупреждение C26110: вызывающая сторона не удерживает блокировку \<lock> перед вызовом функции \<func> .

Если требуется блокировка, обязательно выясните, должна ли сама функция или ее вызывающая сторона получить блокировку. Предупреждение C26110 выдается при нарушении `_Requires_lock_held_` заметки или других заметках, связанных с блокировкой. Дополнительные сведения см. в разделе [Аннотирование режима блокировки](annotating-locking-behavior.md) .

## <a name="example"></a>Пример

В следующем примере создается предупреждение C26110, так как Аннотация `_Requires_lock_held_` в функции указывает на то `LockRequired` , что вызывающий объект `LockRequired` должен получить блокировку перед вызовом `LockRequired` . Без этой заметки `LockRequired` необходимо получить блокировку до того, как она будет обращаться к любым общим данным, защищенным блокировкой.

```cpp
typedef struct _DATA
{
    CRITICAL_SECTION cs;
    int d;
} DATA;

_Requires_lock_held_(p->cs)

void LockRequired(DATA* p)
{
    p->d = 0;
}

void LockNotHeld(DATA* p)
{
    LockRequired(p); // Warning C26110
}
```
